lib/cw-core/src/util.rs

```diff
@@ -57,3 +57,323 @@ pub fn split_framebuffer(framebuffer: &[u8]) -> Vec<Vec<u8>> {
 
     framebuffers
 }
+
+pub fn get_different_framebuffer(previous_framebuffer: &mut [u8], framebuffer: &[u8]) -> Vec<u8> {
+    //handle incomparable frambuffer
+    if previous_framebuffer.len() != framebuffer.len() {
+        return vec![0]; //error
+    }
+
+    //handle invalid buffer
+    if previous_framebuffer.len() % 3 != 0 {
+        return vec![0];
+    }
```

Comment:
If you would change the function signature to this:

```rust
fn get_different_framebuffer(previous_framebuffer: &[u8; 1200], framebuffer: &[u8; 1200]) -> Vec<u8>
```

the need for the first checks would be gone, since 1200 is always dividable by 3. And length of both would be equal